perception/object_filter/src/detected_object_filter.cpp

```diff
@@ -0,0 +1,192 @@
+
+#include "detected_object_filter/detected_object_filter.hpp"
+
+#include <tier4_autoware_utils/tier4_autoware_utils.hpp>
+
+#include <boost/geometry/algorithms/convex_hull.hpp>
+#include <boost/geometry/algorithms/intersects.hpp>
+
+#include <lanelet2_core/geometry/Polygon.h>
+
+boost::optional<geometry_msgs::msg::Transform> getTransform(
+  const tf2_ros::Buffer & tf_buffer, const std::string & source_frame_id,
+  const std::string & target_frame_id, const rclcpp::Time & time)
+{
+  try {
+    geometry_msgs::msg::TransformStamped self_transform_stamped;
+    self_transform_stamped = tf_buffer.lookupTransform(
+      /*target*/ target_frame_id, /*src*/ source_frame_id, time,
+      rclcpp::Duration::from_seconds(0.5));
+    return self_transform_stamped.transform;
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_WARN_STREAM(rclcpp::get_logger("detected_object_filter"), ex.what());
+    return boost::none;
+  }
+}
+namespace detected_object_filter
+{
+DetectedObjectFilterNode::DetectedObjectFilterNode(const rclcpp::NodeOptions & node_options)
+: Node("detected_object_filter_node", node_options),
+  tf_buffer_(this->get_clock()),
+  tf_listener_(tf_buffer_)
+{
+  using std::placeholders::_1;
+
+  // Set parameters
+  upper_bound_x_ = declare_parameter<float>("upper_bound_x", 100.0);
+  upper_bound_y_ = declare_parameter<float>("upper_bound_y", 50.0);
+  lower_bound_x_ = declare_parameter<float>("lower_bound_x", 0.0);
+  lower_bound_y_ = declare_parameter<float>("lower_bound_y", -50.0);
+  filter_by_xy_position_ = declare_parameter<bool>("filter_by_xy_position", false);
+
+  // Set publisher/subscriber
+  map_sub_ = this->create_subscription<autoware_auto_mapping_msgs::msg::HADMapBin>(
+    "input/vector_map", rclcpp::QoS{1}.transient_local(),
+    std::bind(&DetectedObjectFilterNode::mapCallback, this, _1));
+  object_sub_ = this->create_subscription<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "input/object", rclcpp::QoS{1}, std::bind(&DetectedObjectFilterNode::objectCallback, this, _1));
+  object_pub_ = this->create_publisher<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "output/object", rclcpp::QoS{1});
+}
+
+void DetectedObjectFilterNode::mapCallback(
+  const autoware_auto_mapping_msgs::msg::HADMapBin::ConstSharedPtr map_msg)
+{
+  lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(*map_msg, lanelet_map_ptr_);
+  const lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(lanelet_map_ptr_);
+  road_lanelets_ = lanelet::utils::query::roadLanelets(all_lanelets);
+}
+
+void DetectedObjectFilterNode::objectCallback(
+  const autoware_auto_perception_msgs::msg::DetectedObjects::ConstSharedPtr input_msg)
+{
+  // Guard
+  if (object_pub_->get_subscription_count() < 1) return;
+
+  autoware_auto_perception_msgs::msg::DetectedObjects output_object_msg;
+  output_object_msg.header = input_msg->header;
+
+  if (filter_by_xy_position_) {
```

Comment:
pls add the option only unknown object and the default is true

Reply:
Made unknown only option https://github.com/autowarefoundation/autoware.universe/pull/1221/commits/f80a296e3c6a156d4cb021b7b7aaaa82a32003d6